crypto/fipsmodule/ml_kem/ml_kem_ref/kem.c

```diff
@@ -34,6 +49,11 @@ int crypto_kem_keypair_derand(ml_kem_params *params,
   hash_h(sk+params->secret_key_bytes-2*KYBER_SYMBYTES, pk, params->public_key_bytes);
   /* Value z for pseudo-random output on reject */
   memcpy(sk+params->secret_key_bytes-KYBER_SYMBYTES, coins+KYBER_SYMBYTES, KYBER_SYMBYTES);
+
+  // Abort in case of PCT failure.
+  if (keygen_pct(params, pk, sk)) {
+    BORINGSSL_FIPS_abort();
```

Comment:
For the non-FIPS tests to pass, the abort will need to be surrounded by `#if defined(AWSLC_FIPS)` as in https://github.com/aws/aws-lc/blob/main/crypto/fipsmodule/ec/ec_key.c#L549.